### PR TITLE
Replace Hajimari with Homepage

### PR DESCRIPTION
**Pull request by LilDrunkenSmurf**
Addresses #1125

This replaces hajimari with an opinionated template for homepage.

Based on the current configmap, this is what the dashboard would look like (without grafana):
image

https://github.com/LilDrunkenSmurf/k3s-home-cluster/tree/b28b0a406512dbeb8f715e4cbc67e4f153a1f804/kubernetes/test/apps/default/homepage

### CODE REVIEW

File: bootstrap/templates/addons/homepage/app/configmap.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
```
Comment (onedr0p) [inline bootstrap/templates/addons/homepage/app/configmap.yaml.j2:1]: Maybe use a configmapGenerator for this file instead?

File: bootstrap/templates/addons/kube-prometheus-stack/app/helmvalues.yaml.j2 [hunk 2]
```diff
@@ -94,8 +94,11 @@ data:
         enabled: true
         ingressClassName: internal
         annotations:
-          hajimari.io/appName: Prometheus
-          hajimari.io/icon: simple-icons:prometheus
+          gethomepage.dev/enabled: "true"
+          gethomepage.dev/icon: prometheus.png
+          gethomepage.dev/name: Prometheus
+          gethomepage.dev/group: Observability
+          gethomepage.dev/widget.type: prometheus
```
Comment (onedr0p) [inline bootstrap/templates/addons/kube-prometheus-stack/app/helmvalues.yaml.j2:101]: I wonder if we should start wrapping the annotation blocks with 

```
{% if homepage.enabled | default(false) %}
...
{% endif %}
```

File: bootstrap/templates/addons/homepage/app/helmrelease.yaml.j2 [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      main:
+        ports:
+          http:
+            port: 3000
+    ingress:
+      main:
+        enabled: true
+        className: external
+        annotations:
+          external-dns.alpha.kubernetes.io/target: "external.${SECRET_DOMAIN}"
```
Comment (onedr0p) [inline bootstrap/templates/addons/homepage/app/helmrelease.yaml.j2:59]: Are there any concerns with this being external?

File: bootstrap/templates/addons/kube-prometheus-stack/app/helmvalues.yaml.j2 (hunk 2 above)
Comment (smbonn2005) [reply bootstrap/templates/addons/kube-prometheus-stack/app/helmvalues.yaml.j2:101]: Tend to agree here, just to help keep things a little cleaner, especially if someone decides they don't want to use homepage.

Comment (LilDrunkenSmurf) [reply bootstrap/templates/addons/kube-prometheus-stack/app/helmvalues.yaml.j2:101]: 100% we should do that. I'll add it. Great suggestion.

File: bootstrap/templates/addons/homepage/app/helmrelease.yaml.j2 (hunk 3 above)
Comment (LilDrunkenSmurf) [reply bootstrap/templates/addons/homepage/app/helmrelease.yaml.j2:59]: Yeah, this should likely be internal.

File: bootstrap/templates/addons/homepage/app/configmap.yaml.j2 (hunk 1 above)
Comment (LilDrunkenSmurf) [reply bootstrap/templates/addons/homepage/app/configmap.yaml.j2:1]: I think I previously used a configmapGenerator, but it didn't parse the vars, so I needed to drop all my secrets into cluster-secrets, which I didn't like (or put them plaintext in the annotations, which is not great).

We can go either way, but I found this one to be the most consistent.

Comment (onedr0p) [reply bootstrap/templates/addons/homepage/app/configmap.yaml.j2:1]: This is fine then

**Review by onedr0p — APPROVED** [review]

**Pull request merged**